### DISCUSSION TITLE
Components that can't be controlled and depend on useEffect

### DISCUSSION THREAD

**Discussion by davicostalf** [vtex/shoreline#736] [General]
Comentário no CSAT de Engenharia em 2023 H1:

> O que me incomoda mais são os componentes que não são usáveis como controlled components, geralmente alguma limitação no hook do state. Eles acabam precisando serem usados com useEffect pra conseguir o efeito desejado. 
> Não lembro outros casos de uso diretamente, mas acontece mais de precisar de um useEffect no AdminUI do que em outras libs que usei.

**Comment by matheusps** [comment]
Resolvido no Shoreline #1025
